### PR TITLE
Performance improvements and new features

### PR DESCRIPTION
**Pull request by scottwittenburg**
- flatten `sessionTree` data structure
- compute array range when images are loaded into memory (supporting improved performance)
- make sure single arrows don't skip past last (or first) time step to next (or previous() scan
- make sure single arrow buttons (and left/right arrow keys) don't skip any time steps
- press+hold single arrow buttons (and press+hold left/right arrow keys) scan as fast as possible through time steps
- redesign window level controls
- support loading original csv format as well as new json format
- migrate `vue-utilities` code into this repo as it seems now defunct

~~This set of changes requires changes in `vtk.js` ([here](https://github.com/Kitware/vtk-js/pull/1604)) and `vue-utilties` ([here](https://github.com/Kitware/vue-utilities/pull/2)).~~

The `vtk.js` change was merged into master and the `vue-utilities` code was moved into this repo.

### CODE REVIEW

File: client/src/components/WindowControl.vue [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default {
   computed: {
     ...mapState(["proxyManager", "currentDataset"]),
     representation() {
+      console.log("goop");
```
Comment (dzenanz) [inline client/src/components/WindowControl.vue]: perhaps something more descriptive than "goop"? 😄

**Review by dzenanz — COMMENTED** [review]
LGTM

File: client/src/components/WindowControl.vue (hunk 1 above)
Comment (scottwittenburg) [reply client/src/components/WindowControl.vue]: 😆 I like to sprinkle things in like that, just to check if anyone's paying attention.  So good job 😜 

It's definitely a work in progress, I'll make sure to go through at the end and remove debug messages, commented code, and other cruft.  Thanks for taking a look though!

**Review by dzenanz — COMMENTED** [review]
Looks good on a glance.

**Comment by scottwittenburg** [comment]
As we discussed last week, I'm just going to merge this now.

**Pull request merged**